mkimage.py

```diff
@@ -8,6 +8,102 @@
 
 import make
 
+from collections import namedtuple
+
+
+def round(i):
+    return int(i / 4) * 4
+
+
+_Region = namedtuple("Region", ["name", "desc", "start", "size"])
+class Region(_Region):
+    def __init__(self, name, desc, start, size):
+        #_Region.__init__(self, name, start, size, desc)
+        assert round(start) == start, (
+            "Start invalid {} != {}".format(start, round(start)))
+        assert round(size) == size, (
+            "Size invalid {} != {}".format(size, round(size)))
+
+    def __str__(self):
+        return "Region(0x{:06x}, 0x{:06x}, {:10s}, {})".format(
+            self.start, self.end, repr(self.name), repr(self.desc))
+
+    @property
+    def end(self):
+        return self.start + self.size
+
+
+def get_regions(size_gw, size_bios, size_flash):
+    """
+
+    >>> regions = get_regions(256, 128, 1000)
+    >>> for r in regions:
+    ...     print(r)
+    Region(0x000000, 0x000100, 'Gateware', 'FPGA bitstream')
+    Region(0x000100, 0x000180, 'BIOS'    , 'LiteX BIOS with CRC')
+    Region(0x000180, 0x0003e8, 'Firmware', 'Firmware in FBI format')
+
+    """
+    region_gw = Region(
+        "Gateware",
+        "FPGA bitstream",
+        0,
+        round(size_gw),
```

Comment:
Really we should be rounding upwards?

Reply:
That's probably right - I'll rework `round` function to ensure this.